source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -341,14 +449,39 @@ FilterStatus ConnectionManager::ActiveRpc::applyDecoderFilters(ActiveRpcDecoderF
   if (upgrade_handler_) {
     // Divert events to the current protocol upgrade handler.
     const FilterStatus status = filter_action_(upgrade_handler_.get());
+    filter_action_ = nullptr;
     filter_context_.reset();
     return status;
   }
 
-  std::list<ActiveRpcDecoderFilterPtr>::iterator entry =
-      !filter ? decoder_filters_.begin() : std::next(filter->entry());
-  for (; entry != decoder_filters_.end(); entry++) {
-    const FilterStatus status = filter_action_((*entry)->handle_.get());
+  return applyFilters<ActiveRpcDecoderFilter>(filter, decoder_filters_);
+}
+
+FilterStatus
+ConnectionManager::ActiveRpc::applyEncoderFilters(DecoderEvent state, absl::any data,
+                                                  ProtocolConverterSharedPtr protocol_converter,
+                                                  ActiveRpcEncoderFilter* filter) {
+  ASSERT(filter_action_ == nullptr || state == DecoderEvent::Default);
+  prepareFilterAction(state, data);
+
+  if (applyFilters<ActiveRpcEncoderFilter>(filter, encoder_filters_, protocol_converter) !=
+      FilterStatus::Continue) {
+    throw EnvoyException("Not supported.");
```

Comment:
I think we've been trying to get away from throwing in the data path, so let's just `ASSERT()` here.

Reply:
done